cli/tests/test_split_command.rs

```diff
@@ -675,3 +680,199 @@ fn test_split_interactive_with_paths() {
     ◆  zzzzzzzz root() 00000000
     ");
 }
+
+// When a commit is split, the second commit produced by the split becomes the
+// working copy commit for all workspaces whose working copy commit was the
+// target of the split. This test does a split where the target commit is the
+// working copy commit for two different workspaces.
+#[test]
+fn test_split_with_multiple_workspaces_same_working_copy() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    test_env.jj_cmd_ok(&main_path, &["desc", "-m", "first-commit"]);
+    std::fs::write(main_path.join("file1"), "foo").unwrap();
+    std::fs::write(main_path.join("file2"), "foo").unwrap();
+
+    // Create the second workspace and change its working copy commit to match
+    // the default workspace.
+    test_env.jj_cmd_ok(
+        &main_path,
+        &["workspace", "add", "--name", "second", "../secondary"],
+    );
+    // Change the working copy in the second workspace.
```

Comment:
nit: Seems redundant with line 700

Reply:
Yep, thanks. I restructured the test a bit and forgot to delete this.

Reply:
This line is still here. Should be squashed from the child commit.